Update circlesgrid.cpp

grundman opened the pull request:
Use identity matrix if homography finding failed. Current behavior zeros out all points.

<!-- Please use this line to close one or multiple issues when this pullrequest gets merged
You can add another line right under the first one:
resolves #1234
resolves #1235
-->

### This pullrequest changes

<!-- Please describe what your pullrequest is changing -->

sovrasov commented on modules/calib3d/src/circlesgrid.cpp:
```diff
@@ -839,7 +839,7 @@ Mat CirclesGridFinder::rectifyGrid(Size detectedGridSize, const std::vector<Poin
   //Mat H = findHomography( Mat( corners ), Mat( dstPoints ) );
 
   if (H.empty())
-      H = Mat::zeros(3, 3, CV_64FC1);
```
I think in case of `findHomography` failure `warpedKeypoints` are useless at all, so amount of work can be reduced:
```
if (H.empty())
  {
      H = Mat::zeros(3, 3, CV_64FC1);
      warpedKeypoints.clear();
      return H;
  }
```

grundman commented:
addressed comments

sovrasov commented on modules/calib3d/src/circlesgrid.cpp:
```diff
@@ -837,9 +837,13 @@ Mat CirclesGridFinder::rectifyGrid(Size detectedGridSize, const std::vector<Poin
   Mat H = findHomography(Mat(centers), Mat(dstPoints), RANSAC);
   //Mat H = findHomography( Mat( corners ), Mat( dstPoints ) );
-
+ 
```
Please, squash commits into one and delete trailing space.

grundman commented:
addressed comments - can you merge with squash plz? (https://github.com/blog/2141-squash-your-commits) I did this over the web interface and would be nice to avoid checking this out and squashing it locally on my machine.

sovrasov commented:
:+1:

grundman commented:
Can you merge with squash? If not let me know and I squash on my end
